### PR TITLE
Ports - Margaret 

### PR DESCRIPTION
**Pull request by msfinnan**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is where you put the Ruby code that communicated with the database and where the business logic lives. 
Describe in your own words what the Controller is doing in Rails | The controller manages the Rails app. It takes requests from the server, gets information from the models, and makes decisions. Actions/methods are in the controller. 
Describe in your own words what the View is doing in Rails | The view turns data into HTML and does not hold any business logic. 
Describe an edge-case controller test you wrote | I wrote an edge-case controller test for when data does not save during an update. 
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params is a safety measure to regulate which parameters the application takes in. 
How are Rails migrations related to Rails models? | When you generate a migration in the command line (ex. rails generate model task name:string description:string completed:boolean) you are creating the database with the table columns name, description, and completed & their corresponding database. Doing this also creates an empty model called task. 

Describe one area of Rails that are still unclear on | Still wrapping my head around how all the files communicate with each other, which files I need to pay attention to, and which are just running in the background and I can ignore.

### CODE REVIEW

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -77,52 +72,122 @@
...
+
+      # Assert
+      must_respond_with :not_found
+    end
   end
 
-  # Complete for Wave 4
   describe "toggle_complete" do
-    # Your tests go here
+    it "changes completed from true to false" do
```
Comment (CheezItMan) [inline test/controllers/tasks_controller_test.rb:181]: You should also verify that you can toggle the completion date back to false.

File: app/views/tasks/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        <li><%= link_to "All Tasks", tasks_path %></li>
+      </ul>
+    </nav>
+  <div class="all_tasks">
+  <h2>My Tasks</h2>
+  <ol>
+  <% @tasks.each do |task| %>
+    <% if task.completed == false%>
+      <li>
+        <a href="/tasks/<%=task.id%>">
```
Comment (CheezItMan) [inline app/views/tasks/index.html.erb:16]: You should really use `<%= link_to task.task_name, task_path(task.id) %>`

File: app/views/tasks/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <div class="links">
+        <%= link_to "Mark complete", { action: "toggle_complete", id: task.id }, method: :patch %> |
+        <%= link_to "Edit", edit_task_path(task.id) %> |
+        <%= link_to "Delete", task_path(task.id), method: :delete, data: {confirm: "Are you sure you want to delete this book??"}%> 
+        </div>
+      </li>
+      <% else %>
+      <li>
+        <div class="strike-through">
+        <a href="/tasks/<%=task.id%>">
```
Comment (CheezItMan) [inline app/views/tasks/index.html.erb:27]: Another spot to use `link_to`

**Comment by CheezItMan** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | Good number of commits and good commit messages
Answered comprehension questions |  Check, you'll get more familiar with things as you work with Rails
Successfully handles: Index, Show | Check
Index & Show tests pass |Check
Successfully handles: New, Create | Check
New & Create tests pass |Check
Successfully handles: Edit, Update |Check
Tests for Edit & Update test valid & invalid task ids | Check
Successfully handles: Destroy, Task Complete | Check
Tests for Destroy & Task Complete include tests for valid and invalid task ids | Check
Routes follow RESTful conventions | Check
Uses named routes (like `_path`) |**NOOOOO** you need to use link_to and not hardcoded paths. You only have that in a couple of places.
**Overall** | Very well done, you hit all the learning goals for the project.  Nicely done.  Take a look at my _minor_ comments inline in your code and let me know if you have questions.
